Check apostroph tool

hgouraud opened the pull request:
To detect potential conflicts between aaa'bbb and aaa’bbb.

hgouraud commented:
The current version of -key in gwfixbase does not detect potential conflicts due to the various forms of apostrophe.
The problem may be specific to the apostrophe character.

hgouraud commented on bin/fixbase/gwfixbase.ml:
```diff
@@ -178,6 +183,7 @@ let fevents_witnesses = ref false
 let marriage_divorce = ref false
 let invalid_utf8 = ref false
 let key = ref false
+let utf8_key = ref false
```
Fixbase.fix_key does not detect conflicts arising with ' and ‘ and ’ in names.
Fixbase.scan_utf8_conflicts does.
I did not try to merge those two functions, but it may be worth a try

swrup commented on bin/fixbase/gwfixbase.ml:
```diff
@@ -197,6 +203,8 @@ let speclist =
   ; ("-person-key", Arg.Set key, " missing doc")
   ; ("-index", Arg.Set index, " rebuild index. It is automatically enable by any other option.")
   ; ("-invalid-utf8", Arg.Set invalid_utf8, " missing doc")
+  ; ("-key", Arg.Set key, " fix duplicate keys")
```
Humm, I would prefer we keep "-person-key" but with updating it's doc, and remove "-key" so we don't have to change any script running fixbase, and don't have a duplicate parameter

hgouraud committed:
Rewrite special_utf_8, remove -key parameter

hgouraud commented on lib/fixbase.ml:
```diff
@@ -439,3 +439,92 @@ let fix_key ?report progress base =
+           (Char.code s.[2] = 0x98 || (* ’ apostrophes typo *)
+            Char.code s.[2] = 0x99)   (* ‘ autre apostrophe *)
+          -> " "
+    | _ -> s
```
Thanks for the comment. Updated accordingly

swrup commented:
This add a scan that detect aaa'bbb and aaa’bbb as homonyms?
all apostrophe are changed to the same one, so people with same name but different apostrophe are now homonyms?

hgouraud commented:
No, aaa'bbb and aaa‘bbb do not become homonyms as the occ number for aaa‘bbb is changed to avoid this conflict.

swrup commented:
Doesn't changing someone name break lots of things? (links in wiki, portrait file, ... )

hgouraud commented:
Yes, and a warning to the user should be added, similar to the warning that should exist when changing fn, occ, sn for someone whose entry in nldb is not empty.
I have implemented this warning for Roglo, but I need to check if I transposed it in the V7!!
(après vérifications, le code correspondant est dans print_mod_ok, lignes 988 à 1010. A voir comment factoriser ce code et le rendre utilisable par fixbase.

hgouraud commented:
The ideal would be indeed to do the full change automatically!!

Ideally, fixbase should use the same code as UpdateIndOk, and it should be there that all these updates are done.
In UpdateIndOK, the portrait is changed in effective_mod (line 816), and nldb is changed at line 1183, but I suspect this does not change the content of the personnal notes of other persons.

hgouraud commented:
Still some work: 
- rename portraits if pnoc has been changed!
- obtain wizard password to allow opening of gwd pages needing update

hgouraud committed:
remove utf8 stuff from fixbase, put it all in gwfixutf8

redianthus commented on bin/fixbase/gwfixbase.ml:
```diff
@@ -1,6 +1,22 @@
+let index = ref false
+let dry_run = ref false
+let server = ref "localhost"
+let gwd_port = ref 2317
```
I don't understand why you moved all this ?

redianthus commented on bin/fixutf8/dune:
```diff
@@ -0,0 +1,6 @@
+(executable
+  (name gwfixutf8)
+  (public_name geneweb.gwfixutf8)
```
Could we only have a `public_name` ?

redianthus commented on bin/fixutf8/gwfixutf8.ml:
```diff
@@ -0,0 +1,455 @@
+  let normalize_utf_8_date ifam iper s =
+    let s' = Mutil.normalize_utf_8 s in
+    (if s <> s' then
+     match report with
```
`Option.iter` ?

hgouraud committed:
update gwsetup to include gwfixutf8

hgouraud committed:
add multi lines translations in gwsetup (see gwfixutf8.htm)

Halbaroth committed:
Do not fail if we cannot open `notes_d` directory

PR #2105 changes the behavior of GeneWeb with the directory `notes_d`.
The previous iterator silently ignores Unix errors while exploring this
directory. In particular, no error was emitted if the directory
`notes_d` did not exist.

This PR restores this behavior.

hgouraud committed:
fix gwfixutf8 after rebase (wip)

hgouraud committed:
fix ' in Name.lower

hgouraud committed:
hack in Name.lower to handle apostrophes

hgouraud committed:
fmt notesDisplay

hgouraud committed:
fix gwfixutf8 for apostrophes conflict

hgouraud committed:
some fixes after merge

hgouraud committed:
revert hack for apostrophs

hgouraud commented:
Obsolete after unidecode.0.5.0
